xdsl/ir.py

```diff
@@ -1612,3 +1613,129 @@ def is_structurally_equivalent(
         ):
             return False
         return True
+
+
+class _AffineExprKind(Enum):
+    Add = ("+",)
+    Mul = ("*",)
+    Mod = ("mod",)
+    FloorDiv = ("floordiv",)
+    CeilDiv = ("ceildiv",)
+    Constant = ("const",)
+    DimId = ("d",)
+    SymbolId = ("s",)
+
+
+class _AffineExprStorage(ABC):
```

Comment:
Also, I would make these class `@dataclasss`, so you generate directly the `__init__`, and the `__eq__`.